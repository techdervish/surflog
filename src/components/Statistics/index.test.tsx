import {
  createShallowWrapper,
  expectWrappersTreeToMatchSnapshot,
  mocks,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('Statistics component', () => {
  const mockFn = jest.fn();
  const wrapper = createWrapper({
    storage: mocks.storage,
    onClickManage: mockFn,
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('renders count', () => {
    expect(wrapper.find('.total').text()).toEqual('10 records in total');
  });

  it('calls onClickManage on click', () => {
    wrapper.find('a').simulate('click');
    expect(mockFn).toBeCalled();
  });

  it('renders no record when count is 0', () => {
    const noItemWrapper = createWrapper({ storage: { count: 0 } });
    expect(noItemWrapper.find('.__t')).toHaveLength(1);
  });
});
