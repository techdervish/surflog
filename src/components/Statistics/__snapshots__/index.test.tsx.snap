// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics component renders the expected snapshot 1`] = `
<div
  className="tc f4 fw3 ph3 flex pv3 justify-center"
>
  <div
    className="total"
  >
    10 records in total
  </div>
  ,   
  <a
    className="fw2 bg-animate black bb"
    onClick={[MockFunction]}
    style={
      Object {
        "cursor": "pointer",
        "padding": "0 4px",
      }
    }
  >
      
    manage records
  </a>
</div>
`;
